Doc Browser Export & Customisation

gampleman opened the pull request:
This PR allows the doc browser to be customised and exported by the user.

### Documentation

Documentation updates that describe how this works are here: https://github.com/rightscale/praxis/pull/133

### Changes

First step in some of the changes proposed in #92. Implements the following changes:

1. All doc browser stuff is now centralised under the `praxis:docs` namespace.
2. The doc browser requires node.js. (TODO: add this to the docs PR)
3. `rake praxis:docs:generate` generates the JSON schema files. `rake praxis:api_docs` is now an alias of this with the idea that `rake praxis:api_docs` will be deprecated.
4. `rake praxis:docs:preview` will open a browser window with the doc browser. It will refresh automatically when the design files change as well as when any customisations change.
5. `rake praxis:docs:build` will generate a fully built static version of the app in `docs/output`. This can then be put on S3 or GH pages and should work.
6. The default app generator will now generate a docs directory with some files to get you started.
7. Any `.js` file in the `docs` directory will be included in the doc browser. Angular's dependency injection allows users to override any service or controller as needed.
8. The default `docs/styles.scss` simply `@import 'praxis'`. However this means the user can override any of bootstraps variables allowing for easy theme customisation. Rules etc. can now also be overridden.
9. Any templates defined in `docs/views` take precedence over those defined in the doc browser. This means the user can easily customise parts of the app, while leaving the rest up to us.
10. Any changes to the above customisations will be automatically reloaded on save.

### Example

You can see some of these customization options and the export [here](http://praxis-stuff2.s3.amazonaws.com/index.html):

[![screen shot 2015-03-05 at 15 38 41](https://cloud.githubusercontent.com/assets/69144/6507991/c2c6e328-c34d-11e4-8b99-9586be2e1709.png)](http://praxis-stuff2.s3.amazonaws.com/index.html)



This is achieved using these customizations:

https://github.com/rightscale/resticle_examples/compare/blogger...custom-blogger?expand=1

blanquer commented:
Looks great @gampleman!
can you please add a new entry to the Changelog for this? you can simply add a bullet saying: Document browser improvements, and simply paste the nice description bullets you have in the PR under it.

Also, sign your commits, please.

gampleman commented:
Do you need to sign merge commits too?

blanquer commented:
oh, shoot, sorry @gampleman , no I don't need signed merge commits. I must have clicked on the wrong one. Sorry, your direct commit is nicely signed yes. 
Just paste the info into the Changelog please.

gampleman commented:
@blanquer I added the changelog and fixes a few issues with the new app generator. 

Also I added a bit into one of the Rake tasks so that it generates the required files if `/docs` doesn't exist. That should make upgrading a bit smoother.

blanquer commented:
LGTM
